base/reinterpretarray.jl

```diff
@@ -725,11 +725,18 @@ end
 @assume_effects :total function array_subpadding(S, T)
     lcm_size = lcm(sizeof(S), sizeof(T))
     s, t = CyclePadding(S), CyclePadding(T)
-    isempty(t) && return true
-    isempty(s) && return false
     checked_size = 0
-    ps, sstate = iterate(s)   # use of Stateful harms inference and makes this vulnerable to invalidation
-    pad, tstate = iterate(t)
+    # use of Stateful harms inference and makes this vulnerable to invalidation
```

Comment:
This does not seem to use Stateful, but just the normal iterate algorithm?
```suggestion
```

Reply:
I guess it was written with Stateful before and got changed to not use it and this is here as a hint for future developers to not put that back?

Reply:
Correct - it was changed some time before 1.9 by Tim Holy in hunt of invalidations. I left the comment.

Reply:
https://github.com/JuliaLang/julia/pull/47889/files